packages/core/src/components/modal/test/default/modal.e2e.ts

```diff
@@ -0,0 +1,25 @@
+import { test } from 'stencil-playwright';
+import { expect } from '@playwright/test';
+
+const componentTestPath = 'src/components/modal/test/default/index.html';
+
+test.describe('tds-modal', () => {
+  test('renders modal correctly', async ({ page }) => {
+    await page.goto(componentTestPath);
+    const tdsModal = page.getByTestId('tds-modal-testid');
+    await expect(tdsModal).toHaveCount(1);
+    await expect(page).toHaveScreenshot({ maxDiffPixels: 0 });
+  });
+
+  test('modal contains two buttons', async ({ page }) => {
```

Comment:
modal actually contains 3 buttons. also the close button right?

So something like

```
  test('modal contains three buttons', async ({ page }) => {
    await page.goto(componentTestPath);
    // Locate the buttons within the modal
    const modalButtons = page.getByRole('button');

    // Assert that there are three buttons
    await expect(modalButtons).toHaveCount(3);
  });
```

Note that this only finds the elements if the modal is show by default

Reply:
True!